Source/Diagnostics/ReducedDiags/ParticleNumber.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2019-2020 Neil Zaim, Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "ParticleNumber.H"
+#include "WarpX.H"
+
+// constructor
+ParticleNumber::ParticleNumber (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    // get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    // get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
+
+    // get number of species (int)
+    auto nSpecies = mypc.nSpecies();
+
+    // resize data array to nSpecies+1
+    // (number of particles of each species + total number of particles)
+    m_data.resize(nSpecies+1, amrex::Real(0.));
+
+    // get species names (std::vector<std::string>)
+    auto species_names = mypc.GetSpeciesNames();
```

Comment:
Can this be `const` ?
```suggestion
    const auto species_names = mypc.GetSpeciesNames();
```